js/signup.js

```diff
@@ -14,32 +15,60 @@ window.onload = function(){
     const pwError = document.querySelector(".signup__password--error");
     const pwConfirmError = document.querySelector(".signup__password--confirm--error");
 
-    // 회원 가입 함수
-    function register(){
-        // 유효성 검사(정규 표현식)
-        if(!regexEamil.test(emailInput.value)){
-            emailError.textContent = "올바른 이메일 주소가 아닙니다.";
-        }
-        else if(!regexPassword.test(pwInput.value) || !regexPassword.test(pwConfirmInput.value)){
-            pwError.textContent = "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
-        }
-        // 이메일 중복확인, 패스워드 일치 여부
-        else if(userInfo.email === emailInput.value){
-            emailError.textContent = "이미 사용 중인 이메일입니다.";
-        }
-        else if(pwInput.value !== pwConfirmInput.value){
-            pwConfirmError.textContent = "비밀번호가 일치하지 않아요.";
-            
-        }else{
-            location.href = '/folder.html';
+    // 이메일 중복확인 => 회원가입
+    async function register(){
+        try{
+            await vaildEmail(emailInput, emailError);
+            await vaildPW(pwInput, pwError);
+            await vaildConfirmPW(pwConfirmInput, pwConfirmError);
+            await passwordDiffCheck(pwInput, pwConfirmInput);
+            const eamilCheck = {
+                email: emailInput.value
+            }
+            const response = await fetch(API_PATH_CHECK_EMAIL, {
+                method : "POST",
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(eamilCheck)
+            })
+            const result = await response.json();
+            if(response.ok){
```

Comment:
중첩된 조건문이 많기 때문에 이런경우에는 빠른 리턴을 시켜주는게 좋습니다. 

```suggestion
            const response = await fetch(API_PATH_CHECK_EMAIL, {
                method : "POST",
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify(eamilCheck)
            })
            const result = await response.json();
            if(!response.ok){
                return
              }
```